### PR TITLE
[sc-307] Add string syntax to hvml

### PR DESCRIPTION
**Pull request by imaqtkatt**

### CODE REVIEW

File: src/term/net_to_term.rs [hunk 1]
```diff
@@ -207,6 +210,25 @@ impl<'a> Reader<'a> {
   }
 }
 
+fn decode_str(fst: Term, bod: Term) -> Term {
```
Comment (tjjfvi) [inline src/term/net_to_term.rs:213]: These `unreachable!`s should instead push readback errors (as many of them are likely reachable in debug mode)

File: src/term/parser/lexer.rs [hunk 2]
```diff
@@ -33,6 +33,14 @@ pub enum Token {
   #[regex("0[bB][0-1_]+", |lex| from_radix(2, lex).ok())]
   Num(u64),
 
+  #[regex(r#""([^"\\]|\\t|\\u|\\n|\\")*""#, |lex| {
+    let mut s = lex.slice().to_string();
+    s.pop();
+    s.remove(0);
+    s
```
Comment (tjjfvi) [inline src/term/parser/lexer.rs:40]: The escapes allowed in the regex are not normalized here

File: src/term/transform/encode_strs.rs [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+        arg.encode_str();
+      }
+      Term::Tup { fst, snd } | Term::Sup { fst, snd, .. } => {
+        fst.encode_str();
+        snd.encode_str();
+      }
+      Term::Opx { fst, snd, .. } => {
+        fst.encode_str();
+        snd.encode_str();
+      }
```
Comment (tjjfvi) [inline src/term/transform/encode_strs.rs:52]: These branches can be merged

File: src/term/display.rs [hunk 4]
```diff
@@ -67,6 +67,7 @@ impl Term {
       }
       Term::Era => write!(f, "*"),
       Term::Num { val } => write!(f, "{val}"),
+      Term::Str { val } => write!(f, "{val}"),
```
Comment (tjjfvi) [inline src/term/display.rs:70]:
```suggestion
      Term::Str { val } => write!(f, "{val:?}"),
```
(to show it with quotes/escapes)

File: src/term/parser/lexer.rs [hunk 5]
```diff
@@ -33,6 +33,12 @@ pub enum Token {
   #[regex("0[bB][0-1_]+", |lex| from_radix(2, lex).ok())]
   Num(u64),
 
+  #[regex(r#""([^"\\]|\\t|\\u|\\n|\\")*""#, |lex| normalized_string(lex).ok())]
+  Str(String),
+
+  #[regex(r#"'[\W|\w]'"#, num_of_char)]
```
Comment (tjjfvi) [inline src/term/parser/lexer.rs:39]:
```suggestion
  #[regex(r#"'.'"#, num_of_char)]
```

Comment (tjjfvi) [reply src/term/parser/lexer.rs:39]: Although perhaps this should have the same escape handling as strings

**Review by tjjfvi — APPROVED** [review]

**Review by developedby — APPROVED** [review]

**Pull request merged**